tools/DatasetExtraction/dataset.py

```diff
@@ -0,0 +1,37 @@
+from abc import ABCMeta, abstractmethod
+from accessify import private, protected
+import chord
+
+class Dataset:
+    #i - song, j - chord
+    songlist = [[]]
+
+    @protected
+    def validate(self):
+        for song in self.songlist:
+            for ch in song:
+                if not isinstance(ch, chord):
+                    raise Exception("Incorrect chord type")
+                    #will also chech cq_max
+
+
+    #returns songlist with Chord class instances
+    @protected
+    @abstractmethod
+    def load(self):
+        pass
+
+    @private
+    def get_hmm_matrix_i(self):

```

Comment:
this is the best method to start implementation from. Let's implement it's logic stubbing everything else. Once we have this method you can start working on the rest of stuff.